src/Interpreters/Context.cpp

```diff
@@ -2273,7 +2273,8 @@ bool Context::tryCheckClientConnectionToMyKeeperCluster() const
     try
     {
         /// If our server is part of main Keeper cluster
-        if (checkZooKeeperConfigIsLocal(getConfigRef(), "zookeeper"))
+        if (checkZooKeeperConfigIsLocal(getConfigRef(), "zookeeper") || checkZooKeeperConfigIsLocal(getConfigRef(), "keeper")
```

Comment:
you can get config with `zkutil::getZookeeperConfigName(getConfigRef())`
and this check becomes
`if (config_name == "keeper_server" || checkZooKeeperConfigIsLocal(getConfigRef(), config_name))`